c/tskit/trees.c

```diff
@@ -4565,6 +4677,13 @@ tsk_tree_seek(tsk_tree_t *self, double x, tsk_flags_t TSK_UNUSED(options))
         goto out;
     }
 
+    if (self->index == -1) {
+        ret = tsk_tree_seek_from_null(self, x, options);
+        if (ret < 0) {
+            goto out;
+        }
+    }
+
```

Comment:
The rest of the function should all be in an "else" (or maybe a another private utility function, like ``tsk_tree_seek_linear_search``), right? It's not clear that we're not doing the same thing twice (although probably with no effect).